--- a/src/ExtractorFrontend.v
+++ b/src/ExtractorFrontend.v
@@ -1091,8 +1091,8 @@
                     PolIRs.PolyLang.pi_point_witness := PSWIdentity iter_depth;
                     PolIRs.PolyLang.pi_transformation := normalize_affine_list_rev cols tf;
                     PolIRs.PolyLang.pi_access_transformation := normalize_affine_list_rev cols tf;
-                    PolIRs.PolyLang.pi_waccess := normalize_access_list cols w;
-                    PolIRs.PolyLang.pi_raccess := normalize_access_list cols r;
+                    PolIRs.PolyLang.pi_waccess := normalize_access_list (length es) w;
+                    PolIRs.PolyLang.pi_raccess := normalize_access_list (length es) r;
                 |}]
             | None => Err "Instr access extraction/check failed"%string
             end
@@ -1156,8 +1156,8 @@
         PolIRs.PolyLang.pi_point_witness := PSWIdentity iter_depth;
         PolIRs.PolyLang.pi_transformation := normalize_affine_list_rev (env_dim + iter_depth)%nat tf;
         PolIRs.PolyLang.pi_access_transformation := normalize_affine_list_rev (env_dim + iter_depth)%nat tf;
-        PolIRs.PolyLang.pi_waccess := normalize_access_list (env_dim + iter_depth)%nat w;
-        PolIRs.PolyLang.pi_raccess := normalize_access_list (env_dim + iter_depth)%nat r;
+        PolIRs.PolyLang.pi_waccess := normalize_access_list (length es) w;
+        PolIRs.PolyLang.pi_raccess := normalize_access_list (length es) r;
       |}].
 Proof.
     intros instr es constrs env_dim iter_depth sched_prefix pis Hext.
