--- a/VPL/coq/Ring_polynom_AddOnQ.v
+++ b/VPL/coq/Ring_polynom_AddOnQ.v
@@ -208,7 +208,7 @@
       apply Pos2Nat.is_pos.
     + auto.
   - intros; autorewrite with list; auto.
-  Grab Existential Variables. apply O.
+  Unshelve. all: exact O.
 Qed.
 
 Fixpoint bound (pe:PExpr): positive :=
@@ -266,4 +266,4 @@
 Qed.
 
 Theorem PExpr_eq_dec (pe1 pe2: PExpr) : {PExpr_eq pe1 pe2 = true} + {PExpr_eq pe1 pe2 = false}.
-Proof. intuition. Qed.
+Proof. destruct (PExpr_eq pe1 pe2); auto. Qed.
